source/dub/internal/git.d

```diff
@@ -102,6 +91,44 @@ private string determineVersionWithGitTool(NativePath path)
 	return null;
 }
 
+private string determineVersionFromGitDescribe(string describeOutput)
+{
+	import dub.semver : isValidVersion;
+	import std.conv : to;
+
+	const parts = describeOutput.split("-");
+	const commit = parts[$-1];
+	const num = parts[$-2].to!int;
+	const tag = parts[0 .. $-2].join("-");
+	if (tag.startsWith("v") && isValidVersion(tag[1 .. $])) {
+		if (num == 0) return tag[1 .. $];
+		const i = tag.indexOf('+');
+		auto r = format("%s-commit.%s.%s", tag[1 .. (i < 0) ? $ : i], num, commit);
+		if (i > 0) r ~= tag[i .. $];
+		return r;
+	}
+	return null;
+}
+
+unittest {
+	// tag v1.0.0
+	assert(determineVersionFromGitDescribe("v1.0.0-0-deadbeef") == "1.0.0");
```

Comment:
Note that the commit hash will always start with `g`. Don't know if you wish to rely on that or not.